### PR TITLE
feat: add support for custom SVG maps

### PR DESCRIPTION
**Pull request by e0**
For testing locally, please use the https://github.com/MetabolicAtlas/data-files/tree/feat/custom-maps branch in the related data files repository.

### CODE REVIEW

**Comment by nanjiangshu** [comment]
Hi Shan @e0, 

The code looks clear and it works great when I tested. However, I have one concern about how the users can find the link to the custom map. I think it might be better to provide a navigation bar or at least some instructions so that the users know the custom map can be accessed at e.g. http://localhost/explore/Human-GEM/map-viewer/custom_map

**Review by nanjiangshu — COMMENTED** [review]
Hi Shan @e0,

The code looks clear and it works great when I tested. However, I have one concern about how the users can find the link to the custom map. I think it might be better to provide a navigation bar or at least some instructions so that the users know the custom map can be accessed at e.g. http://localhost/explore/Human-GEM/map-viewer/custom_map

**Comment by e0** [comment]
@nanjiangshu Thanks for the review and feedback! I'm not sure if you saw but currently any custom maps should be listed in the sidebar to the left (list of maps). The section is between Compartment maps and Subsystem maps. The `custom_map` in the route is just a test name I have used for the map right now. I think the idea is that eventually there would be several different custom maps, which can be all accessed from the sidebar.

I do agree that we should still provide more ways to guide users to these custom maps. I think it would be best if we can make them in the maps available section in the gem browser pages. That may be out of the scope for the current task but I think we should add a new Jira task for exploring this in more depth.

**Comment by nanjiangshu** [comment]
> @nanjiangshu Thanks for the review and feedback! I'm not sure if you saw but currently any custom maps should be listed in the sidebar to the left (list of maps). 
Sorry I overlooked it. 
 
> I do agree that we should still provide more ways to guide users to these custom maps. I think it would be best if we can make them in the maps available section in the gem browser pages. That may be out of the scope for the current task but I think we should add a new Jira task for exploring this in more depth.

I agree. For this issue I think the current implementation is good enough.

**Review by nanjiangshu — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -153,9 +152,10 @@ const parseModelFiles = (modelDir) => {
   // ========================================================================
   // SVG mapping file
   const svgNodes = [];
-  ['compartment', 'subsystem'].forEach((component) => {
+  ['compartment', 'subsystem', 'custom'].forEach((component) => {
     const filename = `${component}SVG.tsv.plain`;
     const mappingFile = getFile(modelDir, filename);
+    const isCustom = component === 'custom';
```
Comment (mihai-sysbio) [inline index.js:158]: Perhaps it's worth defining a constant for the word `custom`?

**Pull request merged**